src/permutation.jl

```diff
@@ -275,33 +275,54 @@ The `type` parameter specifies use of a two-sided test (`:twosided`) or the dire
 
 See also [`permutation`](@ref).
 
+To account for finite permutations, we implemented the conservative method from Phipson & Smyth 2010:
+ Permutation P-values Should Never Be Zero:Calculating Exact P-values When Permutations Are Randomly Drawn
+ http://www.statsci.org/webguide/smyth/pubs/permp.pdf 
+
 """
-function permutationtest(perm::MixedModelPermutation, model, type::Symbol=:twosided)
+function permutationtest(perm::MixedModelPermutation, model; type::Symbol=:twosided,statistic=:z)
     @warn """This method is known not to be fully correct.
              The interface for this functionality will likely change drastically in the near future."""
 
     if type == :greater || type  == :twosided
-        comp = >
+        comp = >=
     elseif type == :lesser
-        comp = <
+        comp = <=
     else
         throw(ArgumentError("Comparison type $(type) unsupported"))
     end
+    if statistic == :z
+        x = coeftable(model)
+        ests = Dict(Symbol(k) => v for (k,v) in zip(coefnames(model), x.cols[x.teststatcol]))
+    elseif statistic == :β
+        ests = Dict(Symbol(k) => v for (k,v) in zip(coefnames(model), coef(model)))
+    else
+        error("statistic not implemented yet")
+    end
 
-    ests = Dict(Symbol(k) => v for (k,v) in zip(coefnames(model), coef(model)))
-    perms = columntable(perm.β)
+    perms = columntable(perm.coefpvalues)
 
     dd = Dict{Symbol, Vector}()
 
     for k in Symbol.(coefnames(model))
-        dd[k] = perms.β[perms.coefname .== k]
+        dd[k] = perms[statistic][perms.coefname .== k]
+
+        
+        push!(dd[k],ests[k]) # simplest approximation to ensure p is never 0 (impossible for permutation test)
         if type == :twosided
-              μ = mean(dd[k])
-              dd[k] .= abs.(dd[k] .- μ)
-              ests[k] = abs(ests[k].- μ)
+              dd[k]  .= abs.(dd[k])
+              ests[k] = abs(ests[k])
         end
```

Comment:
For tests using beta and the null hypothesis isn't a nil hypothesis, you'll need the mu removal step

Reply:
is \mu guaranteed to be the null-hypothesis? or is it an approximation? we could also loop in the null-beta

Reply:
(Because e.g. H0 => beta = 30, then you'll have values centered 30 and not 0, so the absolute value won't do what you want)

Reply:
I think it makes sense to add a hypothesis vector, especially since you needed one for permutation run.

Reply:
I still think it will be challenging to explain to user, that they get back permutation values for each parameter, but can only interpret one of them that they manually modified...

Maybe the interface should be that you specify one beta, and the others will be automatically set to coef(mod)?

Reply:
That's actually really non trivial from an interface perspective. It might make sense to add a helper function that takes a model and a hypothesis and converts that to a beta vector for the permutation that can be re-used here.

Reply:
agreed. I will implement the beta-H0 thingy analog to the permute function.

For z I don't need to subtract, right?

And finally, the type-I of 0.25 is not resolved yet